scipy/stats/_survival.py

```diff
@@ -374,3 +391,138 @@ def _ecdf_right_censored(sample):
     sf = np.cumprod((n - d) / n)
     cdf = 1 - sf
     return t, cdf, sf, n, d
+
+
+@dataclass
+class LogRankResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+
+
+def log_rank(
+    x: npt.ArrayLike | CensoredData,
+    y: npt.ArrayLike | CensoredData
+) -> LogRankResult:
+    """Compare the survival distributions of two samples via the logrank test.
+
+    Parameters
+    ----------
+    x, y : array_like or CensoredData
+        Samples to compare based on their survival functions.
+
+    Returns
+    -------
+    res : LogRankResult
+        An object containing attributes:
+
+        statistic : float ndarray
+            The computed statistic of the test.
+        pvalue : float ndarray
+            The computed p-value of the test.
+
+    See Also
+    --------
+    scipy.stats.ecdf
+
+    References
+    ----------
+    .. [1] Peto, Richard and Peto, Julian. "Asymptotically Efficient Rank
```

Comment:
Was this reference actually used? If not, I'd remove it. There are many different versions of the test, and it's not immediately obvious that this paper is about the version we have.

Reply:
My reading of other refs was that this was the article that named the test like that. The ordinal article method to be from Mantel.

What I mostly used myself is the website with the example and also some other refs I shared with you. I will add more here.